code/game/objects/structures/tables_racks.dm

```diff
@@ -270,6 +270,9 @@
 		var/obj/item/grab/G = W
 		if(istype(G.grabbed_thing, /mob/living))
 			var/mob/living/M = G.grabbed_thing
+			if(!src.Adjacent(M))
+				visible_message(SPAN_DANGER("[M] is too far to place onto [src]"))
+				return
```

Comment:
I think it would be fine to leave this out for tables. Like you said it's pretty satisfying to do, and it sort of makes sense intuitively that you can "throw"/"teleport" someone around you with a table slam since it's a combat action.